cl/alerts/forms.py

```diff
@@ -6,13 +6,45 @@
 
 from cl.alerts.models import Alert
 from cl.alerts.utils import is_match_all_query
+from cl.search.models import SEARCH_TYPES
 
 
 class CreateAlertForm(ModelForm):
+    ALERT_INCLUSION_CHOICES = [
+        (SEARCH_TYPES.DOCKETS, "Notifications on Cases only"),
+        (SEARCH_TYPES.RECAP, "Notifications on both Cases and Filings"),
+    ]
+    alert_type = forms.ChoiceField(
+        label="What should we include in your alerts?",
+        choices=ALERT_INCLUSION_CHOICES,
+        widget=forms.RadioSelect,
+    )
+
     def __init__(self, *args, **kwargs):
         self.user = kwargs.pop("user", None)
+        initial = kwargs.get("initial", {})
         super().__init__(*args, **kwargs)
 
+        current_type = None
+        # First look at the original alert_type for new alerts / errors.
+        if "original_alert_type" in initial:
+            current_type = initial["original_alert_type"]
+
+        # Otherwise look at the existing instance alert_type for Edit Alert.
+        elif getattr(self, "instance", None) and getattr(
+            self.instance, "pk", None
+        ):
+            current_type = getattr(self.instance, "alert_type", None)
+
+        # Hide the alert_type field for search types other than RECAP.
+        if current_type not in (SEARCH_TYPES.RECAP, SEARCH_TYPES.DOCKETS):
+            self.fields["alert_type"].widget = HiddenInput()
+            # Restore the original valid alert_type field choices, since
+            # they were overridden for RECAP
+            original_field = self._meta.model._meta.get_field("alert_type")
+            field = self.fields["alert_type"]
+            field.choices = original_field.choices
```

Comment:
It looks like the `choices` attribute for the `alert_type` field in the `Alert` model is linked to the value of `SEARCH_TYPES.SUPPORTED_ALERT_TYPES`. So, we could try something like this:

```suggestion
            self.fields["alert_type"].widget = HiddenInput()
            self.fields["alert_type"].choices =  SEARCH_TYPES.SUPPORTED_ALERT_TYPES
```

Reply:
Thanks. Yeah this is simpler!